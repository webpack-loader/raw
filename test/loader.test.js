const path = require('path')
const { webpack } = require('@webpack-utilities/test')

describe('Loader', () => {
  test('Defaults', async () => {
    const config = {
      loader: {
        test: /\.txt$/,
        options: {}
      }
    }

    const stats = await webpack('fixture.js', config)
    const { source } = stats.toJson().modules[1]


    expect(source).toMatchSnapshot()
  })

  describe('Escape', () => {
    test('\\u2028', async () => {
      const config = {
        loader: {
          test: /\.txt$/,
          options: {}
        }
      }

      const stats = await webpack('escape/u2028/fixture.js', config)
      const { source } = stats.toJson().modules[1]


      expect(source).toMatchSnapshot()
    })

    test('\\u2029', async () => {
      const config = {
        loader: {
          test: /\.txt$/,
          options: {}
        }
      }

      const stats = await webpack('escape/u2029/fixture.js', config)
      const { source } = stats.toJson().modules[1]


      expect(source).toMatchSnapshot()
    })
  })
})
