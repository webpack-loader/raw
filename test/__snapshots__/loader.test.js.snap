// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Defaults 1`] = `"export default \\"Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.\\\\n\\""`;

exports[`Loader Escape \\u2028 1`] = `"export default \\"\\\\\\\\u2028\\\\\\\\u2028\\\\n\\""`;

exports[`Loader Escape \\u2029 1`] = `"export default \\"\\\\\\\\u2029\\\\\\\\u2029\\\\n\\""`;
